datamade/openness-project-nmid
# Add option to merge candidates for authenticated users

## Overview

See title. I chose to add a simple form to the candidate detail page, rather than fiddling with the admin interface. This is consistent with the redaction interface for donors.

You can search candidates by name or slug.

Closes #203 

### Demo
Primary object:
![localhost_8001_candidates_bill-b-oneil-8PD1K_](https://github.com/datamade/openness-project-nmid/assets/12176173/8b5ec69d-0fcb-43da-8944-30134ac0bfc5)
Alias object:
![localhost_8001_candidates_bill-b-oneill-dYGI6_](https://github.com/datamade/openness-project-nmid/assets/12176173/e2373b61-851c-4f3d-8521-a1658b4391b8)
Post merge:
![localhost_8001_candidates_bill-b-oneil-8PD1K_ (1)](https://github.com/datamade/openness-project-nmid/assets/12176173/31e7f598-bcd7-4b8e-bb98-adffb961a7c6)

## Testing Instructions

_This test case is modeled in the demo section, above._

* Log into the review app at `/admin` using the credentials datamade / datamade
* From the homepage, search for "Bill B O'neil", then navigate to his detail page.
* In the merge Select2, search for "Bill B O'Neill" (two Ls) and select the corresponding option.
* Submit the form and confirm you see a message that the candidates have been combined. Also confirm that you see multiple campaigns in the right column, and transactions for each campaign at the bottom of the page.
